org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/alloy4/WorkerEngine.java

```diff
@@ -180,8 +183,18 @@ private WorkerEngine() {
     public static void stop() {
         synchronized (WorkerEngine.class) {
             try {
-                if (latest_sub != null)
-                    latest_sub.destroy();
+                if (latest_sub != null) {
+                    if (!System.getProperty("os.name").toLowerCase(Locale.US).startsWith("windows"))
+                        try {  // [HASLab] needed to stop all child processes (electrod)
+                            Field f = latest_sub.getClass().getDeclaredField("pid");
```

Comment:
I don't now the Java Process API, there must a way to get the PID out of a `Process` instance without using reflection to access a private field.

Reply:
not as of java 1.8; either way, I implemented a more elegant workaround to make `WorkerTask`s terminate gracefully (with `destroy()` children processes are not terminated, namely the SMV backends)